websites/permissions.py

```diff
@@ -0,0 +1,163 @@
+        perms (list of str): The permissions to be assigned
+    """
+    for perm in perms:
+        try:
+            assign_perm(perm, group_or_user, website)
+        except Permission.DoesNotExist as err:
+            raise Permission.DoesNotExist(f"Perm {perm} not found") from err
+
+
+@transaction.atomic
+def create_website_groups(website):
+    """
+    Create groups and assign permissions for a website
+
+    Args:
+        website (Website): The website that the permissions are for
+    """
+    admin_group, _ = Group.objects.get_or_create(
+        name=f"{constants.ADMIN_GROUP}{website.uuid.hex}"
+    )
+    editor_group, _ = Group.objects.get_or_create(
+        name=f"{constants.EDITOR_GROUP}{website.uuid.hex}"
+    )
+    assign_object_permissions(admin_group, website, constants.PERMISSIONS_ADMIN)
+    assign_object_permissions(editor_group, website, constants.PERMISSIONS_EDITOR)
+
+
+@transaction.atomic
+def create_global_groups():
+    """Create the global groups for website permissions"""
+    admin_group, _ = Group.objects.get_or_create(name=constants.GLOBAL_ADMIN)
+    author_group, _ = Group.objects.get_or_create(name=constants.GLOBAL_AUTHOR)
+    for perm in constants.PERMISSIONS_ADMIN:
+        assign_perm(perm, admin_group)
+    assign_perm(constants.PERMISSION_ADD, admin_group)
+    assign_perm(constants.PERMISSION_ADD, author_group)
+
+
+def is_global_admin(user):
+    """
+    Determine if the user is a global administrator or superuser
+
+    Args:
+        user (users.models.User): The user to check
+
+    Returns:
+        bool: True if a superuser or member of the global admin group
+
+    """
+    return user.is_superuser or user.groups.filter(name=constants.GLOBAL_ADMIN).exists()
+
+
+def is_site_admin(user, website):
+    """
+    Determine if the user is effectively an admin for a site
+
+    Args:
+        user (users.models.User): The user to check
+        website (Website): The website to check
+
+    Returns:
+        bool: True if user is an admin for the site
+    """
+    return (
+        is_global_admin(user)
+        or website.owner == user
+        or user.groups.filter(name=website.admin_group).exists()
+    )
+
+
+def check_perm(user, permission, website):
+    """
+    Determine if the user has a global or website-specific permission
+
+    Args:
+        user (users.models.User): The user to check
+        permission (str): The permission to check
+        website (Website): The website to check
+
+    """
+    return user.has_perm(permission, website) or user.has_perm(permission)
+
+
+class HasWebsitePermission(BasePermission):
+    """Permission to view/modify/create Website objects"""
+
+    def has_permission(self, request, view):
+        if request.method == "POST":
+            # Only global editors and admins can create new Websites
+            return request.user.has_perm(constants.PERMISSION_ADD)
+        return True
```

Comment:
I think we need to return true if `SAFE_METHODS` and return false otherwise, in case there is a `PATCH` or `DELETE` method for the view. Does that make sense? I'm not too familiar with guardian so I might be missing something

Reply:
I thought `PATCH` and `DELETE` would always end up going through the `has_object_permission` function below, because `has_permission` is used for viewing lists and creating (`POST`) new objects.  But I'm not 100% sure of that, and it wouldn't hurt to add the `SAFE_METHODS` conditional.